### PR TITLE
Enable tabbing among tab list headers.

### PR DESCRIPTION
**Pull request by CSilivestru**
Screen readers and keyboard users aren't currently able to tab through the tab list using the tab key (which is expected behaviour from my understanding). I'd like to add the ability to to specify a 0 tabIndex on a tab and have that override the arrow key functionality for more traditional header navigation through the keyboard.

- This PR allows the code author to decide if they would like to use this (so it's backwards compatible to what is there).
- When used, it allows traditional keyboard navigation with the use of <space> or <enter> to activate that `tabPanel`.
- I've included a `tab_enabled` example to show how it works.

### CODE REVIEW

File: src/components/TabPanel.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ export default class TabPanel extends Component {
     selected: PropTypes.bool, // private
     selectedClassName: PropTypes.string,
     tabId: PropTypes.string, // private
+    tabIndex: PropTypes.string,
```
Comment (danez) [inline src/components/TabPanel.js]: This seems unnecessary

**Comment by afercia** [comment]
Worth noting the expected behavior for keyboard navigation is not tabbing, but using the arrow keys.

This is described in a very clear way in the ARIA authoring practices: https://www.w3.org/TR/wai-aria-practices/#tabpanel see "Keyboard Interaction".

The advantage of this interaction model is that it is possible to tab from the selected tab directly to its tabpanel content, skipping the other tabs. Navigation through tabs must work with arrow keys  only and it is the essence of this ARIA widget interaction.

**Comment by CSilivestru** [comment]
Thanks for bringing that up -- I was actually not aware that the aria spec is that specific in this regard!

This is interesting because this change is born our of our regular user testing where keyboard-heavy users are expecting to be able to tab between tabs. I understand that the aria spec is pretty clear on this but I'm open to suggestions on if we should allow this non-spec interaction to be allowed for those users that are not using aria-activated devices/flows and just normal keyboard navigation.

**Comment by afercia** [comment]
I'd say sticking to the expected interaction detailed on the ARIA authoring practices is the best option. Many ARIA widgets adopt this "mixed Tab/Arrow keys" navigation model. And many ARIA widgets are inspired by, and try to emulate, the keyboard interaction already used in our operating systems.

Consider for example the system preferences panel on macOS (or the Task Manager on Windows). They use tabs, and tabs work exactly in the same way: with arrow keys.

screen shot 2016-07-14 at 14 25 47 copy

**Comment by CSilivestru** [comment]
The tabbing approach does seem better for sighted users and I think this is ok to merge because it preserves the spec-described usability you mentioned earlier. This would just allow for both versions of keyboard use :).

**Comment by afercia** [comment]
Would it be an option to consider to make it optional? Again, tabbing interaction is against the spec and it would make this component not compliant.
We've just adopted it in a pretty large project and if this get merged I'm afraid I'd be forced to reconsider our move.

**Comment by danez** [comment]
Merged in 00a8401d100be0c3ec193f761ed1de4d10ef0642

**Pull request closed without merging**